### ISSUE TITLE
Does autoKeras work for timeseries problems involving LSTM?

### ISSUE THREAD

**Issue by bhomass** [keras-team/autokeras#661]
AFAIK, autoML for deep learning is only good for image recognition problems and uses CNN and its close partner components like various pooling layers. Has any one heard of such automation applied to LSTM networks?

**Comment by cjen07** [comment]
+1

**Comment by jetjodh** [comment]
It can be very easily applied to LSTM networks. I used it for text classification tasks successfully.

**Comment by le-dawg** [comment]
+10. Can't believe that there aren't any people interested in this application to time series!

**Comment by mag1cfrog** [comment]
According to AutoKeras's official website, the function of Time Series Forecasting is coming soon.

**Comment by haifeng-jin** [comment]
The Time Series Forecasting is actually in the master branch. We won't release it until we have fully tested its performance.

**Comment by PascalSchimmler** [comment]
It does work pretty well, better than the lstm baseline in my case. I think the issue with lstm is kind of, that there is not as much to optimize. You might run into some trouble, I had to do minor changes to source code to make padding and maskig work for example

**Comment by empereira** [comment]
hello, what is the status to cover the lstm?

**Comment by haifeng-jin** [comment]
LSTM is supported in RNNBlock.

**Comment by kjk111** [comment]
Hi, 
The code below can built an LSTM model for times-series forecasting: 

model = Sequential()
model.add(LSTM(  **N**  , activation='relu', input_shape=(trainX.shape[1], trainX.shape[2]), return_sequences=True))
model.add(LSTM( **n** , activation='relu', return_sequences=False))
model.add(Dropout(0.2))
model.add(Dense(trainY.shape[1]))

model.compile(optimizer='adam', loss='mse')
model.summary()

Have you an idea how I can use the autokeras to built an LSTM. the idea is to find the best **N** and **n** (see the code).

I thank you in advance for your help.

**Comment by haifeng-jin** [comment]
If you have very specific models and hyperparameters to tune in mind, I suggest you try Keras Tuner.

**Comment by stancsz** [comment]
I'm interested in using autokeras for multivariate use case. (Sequence of N input and 1 output. ) im wondering if this has been done before?

**Comment by haifeng-jin** [comment]
@stancsz I assume the current RegressionHead support it, which means you can use any currently supported regression tasks for multivariate regression.
